source/instances/function_map.js

```diff
@@ -0,0 +1,7 @@
+import curryN from '../curryN';
+
+export default function function_map(f, g) {
+  return curryN(g.length, function() {
```

Comment:
Note [this test](https://github.com/ramda/ramda/blob/1a61378c1811479d9820f13581345a9baacf4129/test/complement.js#L18) fails if we implement this as `f(g(x))`.